panther_docking/src/panther_charging_dock.cpp

```diff
@@ -211,11 +224,12 @@ bool PantherChargingDock::isCharging()
     }
 
     if (wibotic_info_->i_charger > kWiboticChargingCurrentThreshold) {
+      setDockPosePublisherState(lifecycle_msgs::msg::Transition::TRANSITION_DEACTIVATE);
       return true;
     }
   } catch (const opennav_docking_core::FailedToDetectDock & e) {
     RCLCPP_ERROR_STREAM(logger_, "An occurred error while checking if charging: " << e.what());
```

Comment:
This catch only seems to catch one throw. Besides, there are quite a lot of checks in this function. Maybe we could pack some of them into a separate helper function?